src/main/scala/io/iohk/ethereum/network/EtcPeerManagerActor.scala

```diff
@@ -178,24 +185,28 @@ class EtcPeerManagerActor(peerManagerActor: ActorRef, peerEventBusActor: ActorRe
     * @return new peer info with the max block number updated
     */
   private def updateMaxBlock(message: Message)(initialPeerInfo: PeerInfo): PeerInfo = {
-    def update(ns: Seq[BigInt]): PeerInfo = {
-      val maxBlockNumber = ns.fold(0: BigInt) { case (a, b) => if (a > b) a else b }
-      if (maxBlockNumber > appStateStorage.getEstimatedHighestBlock())
-        appStateStorage.putEstimatedHighestBlock(maxBlockNumber)
-
-      if (maxBlockNumber > initialPeerInfo.maxBlockNumber)
-        initialPeerInfo.withMaxBlockNumber(maxBlockNumber)
-      else
+    def update(ns: Seq[(BigInt, ByteString)]): PeerInfo = {
+      if (ns.isEmpty) {
         initialPeerInfo
+      } else {
+        val (maxBlockNumber, maxBlockHash) = ns.maxBy(_._1)
+        if (maxBlockNumber > appStateStorage.getEstimatedHighestBlock())
+          appStateStorage.putEstimatedHighestBlock(maxBlockNumber)
+
+        if (maxBlockNumber > initialPeerInfo.maxBlockNumber)
+          initialPeerInfo.withMaxBlockNumber(maxBlockNumber).withMaxBlockHash(maxBlockHash)
```

Comment:
Shouldn't we use a method that updates both number and hash atomically? I think, there is no case to update only one field

Reply:
I wondered about it, so if there is another person with same concern I will change it.